app/models/submission_information_package_zipper.rb

```diff
@@ -19,7 +19,7 @@ def initialize(sip)
   # This key needs to be unique. By default, ActiveStorage generates a UUID, but since we want a file path for our
   # Archivematica needs, we are generating a key. We handle uniqueness on the `bag_name` side.
   def keygen(sip)
-    "etdsip/#{sip.thesis.accession_number}/#{sip.bag_name}.zip"
+    "etdsip/#{sip.thesis.graduation_year}/#{sip.thesis.graduation_month}-#{sip.thesis.accession_number}/#{sip.bag_name}.zip"
```

Comment:
Because you asked...

I think:
```ruby
 def keygen(sip)
    thesis = sip.thesis
    "etdsip/#{thesis.graduation_year}/#{thesis.graduation_month}-#{thesis.accession_number}/#{sip.bag_name}.zip"
end
```

retains readability and addresses the line length. I'm never one to force a change because a bot whined about it though, so I wouldn't say this is a required change.

Another option I feel is _less_ readable (but you may disagree so I'll share it as well) is:


```ruby
['etdsip',
  sip.thesis.graduation_year,
  "#{sip.thesis.graduation_month}-#{sip.thesis.accession_number}",
  sip.bag_name].join('/')
```

Reply:
Good idea. Thanks!